### PR TITLE
Allow changing RabbitMQ UseExternalAuthMechanism and DisableRemoteCertificateValidation via connection string

### LINKED ISSUE #1743: RabbitMQ transport allow activating ExternalAuthMechanism
Hi

We are currently using RMQ for our NSB transport layer with SSL/TLS activated and external auth via ssl client certificates.
For our NSB services this means we have to call 'UseExternalAuthMechanism' on our transport to enable this.
```
var transportExtensions = endpointConfiguration.UseTransport<RabbitMQTransport>();
transportExtensions.ConnectionString(serviceConfig.ConnectionStrings.Messaging);
transportExtensions.UseExternalAuthMechanism();
```
Would it be possible to provide a setting for ServiceControl.Monitoring so this can be enabled?
Looks like the ConnectionFactory always defaults to 'false' in this project:
https://github.com/Particular/ServiceControl.Monitoring/blob/43f9a6a4532ea9a7251d4f12ed2cbb0c7f079f91/src/ServiceControl.Transports.RabbitMQ/QueueLengthProvider.cs#L142
This would allow us to use client certificates by providing them in the connection string.

Thanks

### PR DESCRIPTION
**Pull request by andreasohlund**
Closes #1743

### CODE REVIEW

**Comment by andreasohlund** [comment]
Tested against my local broker and this works as expected. Should be good for final review

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@ nugets
 deploy
 build32
 binaries
+zip
```
Comment (bording) [inline .gitignore]: Why was this changed?

File: src/ServiceControl.Monitoring.Transports.RabbitMQ/ConnectionFactory.cs [hunk 2]
```diff
@@ -3,15 +3,14 @@
...
-    using global::RabbitMQ.Client;
+   using global::RabbitMQ.Client;
 
     class ConnectionFactory
     {
         readonly global::RabbitMQ.Client.ConnectionFactory connectionFactory;
         readonly object lockObject = new object();
 
-        public ConnectionFactory(ConnectionConfiguration connectionConfiguration, X509CertificateCollection clientCertificates, bool disableRemoteCertificateValidation, bool useExternalAuthMechanism)
+        public ConnectionFactory(ConnectionConfiguration connectionConfiguration, bool disableRemoteCertificateValidation, bool useExternalAuthMechanism)
```
Comment (bording) [inline src/ServiceControl.Monitoring.Transports.RabbitMQ/ConnectionFactory.cs]: I know the original intention of this class was to be an exact copy of the one from the transport, so that's why the `clientCertificates` parameter was there.  Perhaps it would be better to leave it as-is?

File: .gitignore (hunk 1 above)
Comment (andreasohlund) [reply .gitignore]: To avoid including the /zip folder that gets created if you build in release mode

File: src/ServiceControl.Monitoring.Transports.RabbitMQ/ConnectionFactory.cs (hunk 2 above)
Comment (andreasohlund) [reply src/ServiceControl.Monitoring.Transports.RabbitMQ/ConnectionFactory.cs]: Ah, I didn't know that. What was the rationale for that?  Easier copy/paste should the transport change?

Comment (bording) [reply src/ServiceControl.Monitoring.Transports.RabbitMQ/ConnectionFactory.cs]: Yes, that was the primary reason I believe.

Comment (andreasohlund) [reply src/ServiceControl.Monitoring.Transports.RabbitMQ/ConnectionFactory.cs]: Ok, I've reverted the code now

**Comment by bording** [comment]
Looks like this needs to be rebased to resolve a conflict. 

Also:

image

What? How did that happen?

**Comment by andreasohlund** [comment]
Hmm, I wonder if that is GitHub for Windows doing something funky. I’ll sort out the conflict
